var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -111,189 +107,77 @@ class FoamExtend(Package):
     depends_on('scotch~metis+mpi', when='+ptscotch')
     depends_on('metis@5:',         when='+metis')
     depends_on('parmetis',         when='+parmetis')
-    depends_on('parmgridgen',      when='+parmgridgen')
+    # mgridgen is statically linked
+    depends_on('parmgridgen',      when='+parmgridgen', type='build')
```

Comment:
I understand, but it will get linked in via the OpenFOAM build logic. In this case spack is reduced to feeding the OpenFOAM config with appropriate paths of where to find things and the OpenFOAM build manages the rest.